rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/NodesRecoveryManager.java

```diff
@@ -75,8 +99,202 @@ public NodesLockRestorationManager apply(RMCore rmCore) {
         };
     }
 
-    public void restoreLocks(RMNode rmNode, Client caller) {
-        nodesLockRestorationManager.handle(rmNode, caller);
+    protected void recoverNodeSourcesAndNodes() {
+        Collection<NodeSourceData> nodeSources = this.rmCore.getDbManager().getNodeSources();
+        this.logPersistedNodeSourceInfo(nodeSources);
+
+        for (NodeSourceData nodeSourceData : nodeSources) {
+            String nodeSourceName = nodeSourceData.getName();
+            if (NodeSource.DEFAULT_LOCAL_NODES_NODE_SOURCE_NAME.equals(nodeSourceName)) {
+                // will be recreated by SchedulerStarter
+                this.rmCore.getDbManager().removeNodeSource(nodeSourceName);
+            } else {
+                this.recoverNodeSourceSuccessfullyOrRemove(nodeSourceData, nodeSourceName);
+            }
+        }
+    }
+
+    protected boolean recoverFullyDeployedInfrastructureOrReset(String nodeSourceName, NodeSource nodeSourceToDeploy,
+            NodeSourceDescriptor descriptor) {
+        boolean recoverNodes = false;
+        boolean existPersistedNodes = this.existPersistedNodes(nodeSourceName);
+
+        if (existPersistedNodes) {
+            InfrastructureManager im = InfrastructureManagerFactory.recover(descriptor);
+            if (!im.getDeployingAndLostNodes().isEmpty()) {
```

Comment:
I see a problem here, we will never recover when there are lost nodes, which were subsequently correctly acquired, is it what we want ?

Reply:
Correct me if I am wrong but I think a node can be lost only if it is deploying first. It is like a 'deploying node that failed to be acquired'. There is no return from the lost state. However, if a node is correctly acquired and then encounters an issue, then it becomes down. 

Thus, if we decide to not recover when there are deploying nodes, then to me it seems natural to also not recover when there are lost nodes.

Reply:
after discussion with @fviale, decided to handle this not in this PR